### PR TITLE
[Ready]Adds hypnosis and the hypnotic stupor brain trauma

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: Added a new severe brain trauma: hypnotic stupor. Victims of hypnotic stupor occasionally fall into a trance, and upon hearing a sentence they'll focus on it to the point of obsession, until it is replaced by a new hypnosis or the trauma is cured.
/:cl:

Adds the potential for some fancy rp gameplay. The idea is that they should take a passing message way too seriously, while leaving room for interpretation to the player. Hearing an isolated "greytide" over the radio might prompt a player to take up arms against the rampaging greyshirts, another might interpret it as having to _become_ the greytide. Someone who realizes what's up might use this to give clear orders to the victim, although they'll only last until the following trance.

I plan on adding more methods to inflict hypnosis with the same trance->"focus on a sentence" mechanics, perhaps as a traitor/CMO pendulum item, or as a wizard/cult spell.

### CODE REVIEW

**Comment by Ratvar** [comment]
"Occasionally you can antag, just remember to say ;VIVA LA REVOLUTION once in a while to help other hypnodudes!"

**Comment by XDTM** [comment]
Honestly if you can get more than one person to roll this specific severe trauma at the same time _and_ collaborate you probably deserve it

**Comment by ExcessiveUseOfCobblestone** [comment]
This probably should have headmin input, if admins are going to ban for interpreting in a way that lets you beat someone up/kill then there's really no point in including it.

**Comment by deathride58** [comment]
This would need *extremely* careful and tight moderation on SFW servers to ensure the feature doesn't get abused for sexual purposes. The erotic hypno community is staggeringly big compared to the vore community, so you're basically guaranteed to find *at least* one person that'll find a "creative" use for this feature.

File: code/datums/brain_damage/hypnosis.dm [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	if(!phrase)
+		qdel(src)
+	..()	
+	
+/datum/brain_trauma/hypnosis/on_gain()
+	message_admins("[ADMIN_LOOKUPFLW(owner)] was hypnotized with the phrase '[hypnotic_phrase]'.")
+	log_game("[key_name(owner)] was hypnotized with the phrase '[hypnotic_phrase]'.")
+	to_chat(owner, "<span class='big'>[hypnotic_phrase]</span>")
+	to_chat(owner, "<span class='notice'>[pick("You feel your thoughts focusing on this phrase... you can't seem to get it out of your head.",\
+												"Your head hurts, but this is all you can think of. It must be vitally important.",\
```
Comment (deathride58) [inline code/datums/brain_damage/hypnosis.dm]: Trances often don't induce migraines on their own.

Comment (XDTM) [reply code/datums/brain_damage/hypnosis.dm]: Yeah i'm using a fairly inaccurate version of hypnosis here, i didn't prioritize realism

**Comment by Arianya** [comment]
May be worth making this an admin message on top of the logging for it, depending on frequency of trances.

**Comment by XDTM** [comment]
Trance is now a status effect, so it can easily be reused for other mechanics (maybe vampires?) in the future.

**Comment by Kagus** [comment]
"Has anyone really been far even as decided to use even go want to do look more like?"

Also yeah, this looks like it could potentially be a bigger source of self-antagging than sillycone laws. Could be fun though!

**Comment by YakumoChen** [comment]
"Kill me" lol

**Comment by XDTM** [comment]
example
Made the highlighting a bit fancier

**Comment by ExcessiveUseOfCobblestone** [comment]
The red bit should be incorporated into the rules, I don't think it's necessary to include everytime you get hypnotized.

**Comment by XDTM** [comment]
The fluff text can be misinterpreted, and hypnosis would be rare enough that it might not be recognizable as such. Since i'll add antag hypnosis later, it would be a big letdown if the guy you hypnotized just ignored the hypnosis because it was just a fluff message to them.

**Comment by ExcessiveUseOfCobblestone** [comment]
Then make the notice text below the big and purple word more obvious that it's a requirement instead of an option.

`<span class='notice'>The phrase constantly dances in your mind. <b>Follow</b> the phrase to the best of your ability.</span>`

Moving it to the rules won't change someone who isn't going to obey the text. It's a feature that relies solely on the other player's willingness so it's going to have the issue of some people breaking that regardless.

**Comment by XDTM** [comment]
I think the reduction in misunderstandings is worth keeping the text. It's not about deliberate rule-breakers, it's about people who think that it's a fluff effect, that it's already enforced mechanically, or who aren't sure they are allowed to be antag-like in following the phrase. Making it clear that it's an rp-based "rule" should prevent all three.

**Comment by ExcessiveUseOfCobblestone** [comment]
Fair enough, just seems like a pain to get the same long message every time you get hypnotized after you already know it enables you to "expand your creativity"

**Comment by Kagus** [comment]
Never underestimate the concentrated stupidity of SS13 players.

**Pull request merged**